app/code/Magento/Ui/view/base/web/js/form/element/post-code.js

```diff
@@ -20,6 +20,26 @@ define([
             }
         },
 
+        /**
+         * Initializes observable properties of instance
+         *
+         * @returns {Abstract} Chainable.
+         */
+        initObservable: function () {
+            this._super();
+
+            /**
+             * equalityComparer function
+             *
+             * @returns boolean.
+             */
+            this.value.equalityComparer = function (oldValue, newValue) {
+                return !oldValue && !newValue || oldValue === newValue;
```

Comment:
Why do we need to change comparator behavior to achieve

>  the compare on the frontend is breaking because NULL != " "

?

Reply:
It exactly says what it is for. We get a NULL value, which is not the same as "". So that's why we need this change.

Reply:
@dverkade I mean this looks like kinda low-level change. Is there some other place in client code which may be adjusted instead of changing comparator behavior?